src/core/net/srp_server.cpp

```diff
@@ -1691,6 +1694,7 @@ Server::UpdateMetadata::UpdateMetadata(Instance &               aInstance,
     : InstanceLocator(aInstance)
     , mExpireTime(TimerMilli::GetNow() + kDefaultEventsHandlerTimeout)
     , mDnsHeader(aHeader)
+    , mId(Random::NonCrypto::GetUint32())
```

Comment:
Do we need to worry about possible duplicate (already in use) IDs (or is it very unlikely)?

Reply:
Do we need the ID to be random? If I am not mistaken, we may only care that it is different per host item
Does this sound correct? If so, can we then consider using an increasing counter for ID? (this way it will take a long time for it to wrap around and reuse the same value).

Reply:
Yeah, I was assuming that there are rare chance for two outstanding updates to have conflict IDs. I chose random ID just because it needs less change :)
But I would agree incremental ID is more suitable and faster though it needs to maintain the `ID` in `Srp::Server`.